src/components/sidebar.js

```diff
@@ -0,0 +1,45 @@
+import newElement from '../rendering/newelement';
+import listElements from '../rendering/listelements';
+import nestElements from '../rendering/nestelements';
+import projectDisplay from './projectdisplay';
+import mainContainer from './maincontainer';
+// eslint-disable-next-line import/no-cycle
+import newProject from '../pages/newproject';
```

Comment:
Generally a dependency cycle is considered bad design in software development, you really want to look into this as opposed to just silencing the linter warning

check out this link for more insight [see](https://stackoverflow.com/questions/51094117/dependency-cycle-detected-import-no-cycle)